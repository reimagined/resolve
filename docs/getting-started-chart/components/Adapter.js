import React, { useState } from 'react'

const Adapter = ({ selected, onClick }) => (
  <g id="interactscm-esadapter" className="box-inner" data-selected={selected}>
    <a onClick={onClick}>
      <path className="box" d="M203 168L267 221.5L203 275L139 221.5L203 168Z" />
      <path
        className="caption"
        d="M182.63 209H189.994V207.489H184.386V203.926H189.545V202.42H184.386V198.875H189.925V197.364H182.63V209ZM199.6 200.273H197.776L195.572 206.989H195.481L193.271 200.273H191.447L194.617 209H196.435L199.6 200.273ZM204.646 209.176C206.549 209.176 207.896 208.239 208.282 206.818L206.674 206.528C206.367 207.352 205.629 207.773 204.663 207.773C203.208 207.773 202.231 206.83 202.185 205.148H208.39V204.545C208.39 201.392 206.504 200.159 204.526 200.159C202.094 200.159 200.492 202.011 200.492 204.693C200.492 207.403 202.072 209.176 204.646 209.176ZM202.191 203.875C202.259 202.636 203.157 201.562 204.538 201.562C205.856 201.562 206.719 202.54 206.725 203.875H202.191ZM211.974 203.818C211.974 202.426 212.826 201.631 214.008 201.631C215.161 201.631 215.86 202.386 215.86 203.653V209H217.559V203.449C217.559 201.29 216.371 200.159 214.587 200.159C213.275 200.159 212.417 200.767 212.013 201.693H211.906V200.273H210.275V209H211.974V203.818ZM223.911 200.273H222.121V198.182H220.423V200.273H219.144V201.636H220.423V206.79C220.417 208.375 221.627 209.142 222.968 209.114C223.508 209.108 223.871 209.006 224.07 208.932L223.763 207.528C223.65 207.551 223.44 207.602 223.167 207.602C222.616 207.602 222.121 207.42 222.121 206.438V201.636H223.911V200.273ZM190.117 216.42H191.81C191.759 214.557 190.112 213.205 187.748 213.205C185.413 213.205 183.623 214.54 183.623 216.545C183.623 218.165 184.782 219.114 186.651 219.619L188.026 219.994C189.271 220.324 190.231 220.733 190.231 221.767C190.231 222.903 189.146 223.653 187.651 223.653C186.299 223.653 185.174 223.051 185.072 221.784H183.31C183.424 223.892 185.055 225.193 187.663 225.193C190.396 225.193 191.969 223.756 191.969 221.784C191.969 219.688 190.1 218.875 188.623 218.511L187.487 218.216C186.577 217.983 185.367 217.557 185.373 216.443C185.373 215.455 186.276 214.722 187.708 214.722C189.043 214.722 189.992 215.347 190.117 216.42ZM198.005 216.273H196.215V214.182H194.516V216.273H193.238V217.636H194.516V222.79C194.511 224.375 195.721 225.142 197.062 225.114C197.602 225.108 197.965 225.006 198.164 224.932L197.857 223.528C197.744 223.551 197.533 223.602 197.261 223.602C196.71 223.602 196.215 223.42 196.215 222.438V217.636H198.005V216.273ZM203.467 225.176C205.927 225.176 207.535 223.375 207.535 220.676C207.535 217.96 205.927 216.159 203.467 216.159C201.006 216.159 199.398 217.96 199.398 220.676C199.398 223.375 201.006 225.176 203.467 225.176ZM203.472 223.75C201.864 223.75 201.114 222.347 201.114 220.67C201.114 219 201.864 217.58 203.472 217.58C205.069 217.58 205.819 219 205.819 220.67C205.819 222.347 205.069 223.75 203.472 223.75ZM209.431 225H211.13V219.67C211.13 218.528 212.011 217.705 213.215 217.705C213.567 217.705 213.965 217.767 214.102 217.807V216.182C213.931 216.159 213.596 216.142 213.38 216.142C212.357 216.142 211.482 216.722 211.164 217.659H211.073V216.273H209.431V225ZM219.021 225.176C220.924 225.176 222.271 224.239 222.657 222.818L221.049 222.528C220.742 223.352 220.004 223.773 219.038 223.773C217.583 223.773 216.606 222.83 216.56 221.148H222.765V220.545C222.765 217.392 220.879 216.159 218.901 216.159C216.469 216.159 214.867 218.011 214.867 220.693C214.867 223.403 216.447 225.176 219.021 225.176ZM216.566 219.875C216.634 218.636 217.532 217.562 218.913 217.562C220.231 217.562 221.094 218.54 221.1 219.875H216.566ZM174.535 241L175.603 237.92H180.148L181.211 241H183.075L178.887 229.364H176.859L172.671 241H174.535ZM176.114 236.443L177.83 231.477H177.921L179.637 236.443H176.114ZM187.745 241.17C189.336 241.17 189.961 240.199 190.268 239.642H190.41V241H192.069V229.364H190.37V233.688H190.268C189.961 233.148 189.381 232.159 187.756 232.159C185.648 232.159 184.097 233.824 184.097 236.653C184.097 239.477 185.626 241.17 187.745 241.17ZM188.12 239.722C186.603 239.722 185.813 238.386 185.813 236.636C185.813 234.903 186.586 233.602 188.12 233.602C189.603 233.602 190.398 234.812 190.398 236.636C190.398 238.472 189.586 239.722 188.12 239.722ZM197.033 241.193C198.477 241.193 199.289 240.46 199.613 239.807H199.681V241H201.34V235.205C201.34 232.665 199.34 232.159 197.954 232.159C196.374 232.159 194.92 232.795 194.352 234.386L195.948 234.75C196.198 234.131 196.835 233.534 197.977 233.534C199.073 233.534 199.636 234.108 199.636 235.097V235.136C199.636 235.756 198.999 235.744 197.431 235.926C195.778 236.119 194.085 236.551 194.085 238.534C194.085 240.25 195.374 241.193 197.033 241.193ZM197.403 239.83C196.442 239.83 195.749 239.398 195.749 238.557C195.749 237.648 196.556 237.324 197.539 237.193C198.09 237.119 199.397 236.972 199.641 236.727V237.852C199.641 238.886 198.817 239.83 197.403 239.83ZM203.603 244.273H205.302V239.642H205.404C205.711 240.199 206.336 241.17 207.927 241.17C210.04 241.17 211.575 239.477 211.575 236.653C211.575 233.824 210.018 232.159 207.91 232.159C206.29 232.159 205.705 233.148 205.404 233.688H205.262V232.273H203.603V244.273ZM205.268 236.636C205.268 234.812 206.063 233.602 207.546 233.602C209.086 233.602 209.859 234.903 209.859 236.636C209.859 238.386 209.063 239.722 207.546 239.722C206.086 239.722 205.268 238.472 205.268 236.636ZM217.552 232.273H215.762V230.182H214.063V232.273H212.785V233.636H214.063V238.79C214.058 240.375 215.268 241.142 216.609 241.114C217.148 241.108 217.512 241.006 217.711 240.932L217.404 239.528C217.29 239.551 217.08 239.602 216.808 239.602C216.256 239.602 215.762 239.42 215.762 238.438V233.636H217.552V232.273ZM223.099 241.176C225.002 241.176 226.349 240.239 226.735 238.818L225.127 238.528C224.82 239.352 224.082 239.773 223.116 239.773C221.661 239.773 220.684 238.83 220.638 237.148H226.843V236.545C226.843 233.392 224.957 232.159 222.979 232.159C220.548 232.159 218.945 234.011 218.945 236.693C218.945 239.403 220.525 241.176 223.099 241.176ZM220.644 235.875C220.712 234.636 221.61 233.562 222.991 233.562C224.309 233.562 225.173 234.54 225.178 235.875H220.644ZM228.728 241H230.427V235.67C230.427 234.528 231.308 233.705 232.512 233.705C232.864 233.705 233.262 233.767 233.398 233.807V232.182C233.228 232.159 232.893 232.142 232.677 232.142C231.654 232.142 230.779 232.722 230.461 233.659H230.37V232.273H228.728V241Z"
      />
    </a>
  </g>
)

export default Adapter
