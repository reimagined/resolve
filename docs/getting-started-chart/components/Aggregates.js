import React, { useState } from 'react'

const Aggregates = ({ selected, onClick }) => (
  <g data-selected={selected}>
    <a onClick={onClick}>
      <path d="M140 89H400V122H140V89Z" fill="#2464C4" />
      <path
        d="M227.339 111L228.408 107.92H232.953L234.016 111H235.879L231.692 99.3636H229.663L225.476 111H227.339ZM228.919 106.443L230.635 101.477H230.726L232.442 106.443H228.919ZM240.942 114.455C243.163 114.455 244.879 113.438 244.879 111.193V102.273H243.214V103.688H243.089C242.788 103.148 242.186 102.159 240.561 102.159C238.453 102.159 236.902 103.824 236.902 106.602C236.902 109.386 238.487 110.869 240.55 110.869C242.152 110.869 242.771 109.966 243.078 109.409H243.186V111.125C243.186 112.494 242.249 113.085 240.959 113.085C239.544 113.085 238.993 112.375 238.692 111.875L237.232 112.477C237.692 113.545 238.857 114.455 240.942 114.455ZM240.925 109.46C239.408 109.46 238.618 108.295 238.618 106.58C238.618 104.903 239.391 103.602 240.925 103.602C242.408 103.602 243.203 104.812 243.203 106.58C243.203 108.381 242.391 109.46 240.925 109.46ZM250.817 114.455C253.038 114.455 254.754 113.438 254.754 111.193V102.273H253.089V103.688H252.964C252.663 103.148 252.061 102.159 250.436 102.159C248.328 102.159 246.777 103.824 246.777 106.602C246.777 109.386 248.362 110.869 250.425 110.869C252.027 110.869 252.646 109.966 252.953 109.409H253.061V111.125C253.061 112.494 252.124 113.085 250.834 113.085C249.419 113.085 248.868 112.375 248.567 111.875L247.107 112.477C247.567 113.545 248.732 114.455 250.817 114.455ZM250.8 109.46C249.283 109.46 248.493 108.295 248.493 106.58C248.493 104.903 249.266 103.602 250.8 103.602C252.283 103.602 253.078 104.812 253.078 106.58C253.078 108.381 252.266 109.46 250.8 109.46ZM257.033 111H258.732V105.67C258.732 104.528 259.612 103.705 260.817 103.705C261.169 103.705 261.567 103.767 261.703 103.807V102.182C261.533 102.159 261.197 102.142 260.982 102.142C259.959 102.142 259.084 102.722 258.766 103.659H258.675V102.273H257.033V111ZM266.622 111.176C268.526 111.176 269.872 110.239 270.259 108.818L268.651 108.528C268.344 109.352 267.605 109.773 266.639 109.773C265.185 109.773 264.207 108.83 264.162 107.148H270.366V106.545C270.366 103.392 268.48 102.159 266.503 102.159C264.071 102.159 262.469 104.011 262.469 106.693C262.469 109.403 264.048 111.176 266.622 111.176ZM264.168 105.875C264.236 104.636 265.134 103.562 266.514 103.562C267.832 103.562 268.696 104.54 268.702 105.875H264.168ZM275.911 114.455C278.132 114.455 279.848 113.438 279.848 111.193V102.273H278.183V103.688H278.058C277.757 103.148 277.155 102.159 275.53 102.159C273.422 102.159 271.871 103.824 271.871 106.602C271.871 109.386 273.456 110.869 275.518 110.869C277.121 110.869 277.74 109.966 278.047 109.409H278.155V111.125C278.155 112.494 277.217 113.085 275.928 113.085C274.513 113.085 273.962 112.375 273.661 111.875L272.2 112.477C272.661 113.545 273.825 114.455 275.911 114.455ZM275.893 109.46C274.376 109.46 273.587 108.295 273.587 106.58C273.587 104.903 274.359 103.602 275.893 103.602C277.376 103.602 278.172 104.812 278.172 106.58C278.172 108.381 277.359 109.46 275.893 109.46ZM284.666 111.193C286.109 111.193 286.922 110.46 287.246 109.807H287.314V111H288.973V105.205C288.973 102.665 286.973 102.159 285.587 102.159C284.007 102.159 282.553 102.795 281.984 104.386L283.581 104.75C283.831 104.131 284.467 103.534 285.609 103.534C286.706 103.534 287.268 104.108 287.268 105.097V105.136C287.268 105.756 286.632 105.744 285.064 105.926C283.411 106.119 281.717 106.551 281.717 108.534C281.717 110.25 283.007 111.193 284.666 111.193ZM285.036 109.83C284.075 109.83 283.382 109.398 283.382 108.557C283.382 107.648 284.189 107.324 285.172 107.193C285.723 107.119 287.03 106.972 287.274 106.727V107.852C287.274 108.886 286.45 109.83 285.036 109.83ZM295.31 102.273H293.52V100.182H291.821V102.273H290.543V103.636H291.821V108.79C291.815 110.375 293.026 111.142 294.366 111.114C294.906 111.108 295.27 111.006 295.469 110.932L295.162 109.528C295.048 109.551 294.838 109.602 294.565 109.602C294.014 109.602 293.52 109.42 293.52 108.438V103.636H295.31V102.273ZM300.857 111.176C302.76 111.176 304.107 110.239 304.493 108.818L302.885 108.528C302.578 109.352 301.839 109.773 300.874 109.773C299.419 109.773 298.442 108.83 298.396 107.148H304.601V106.545C304.601 103.392 302.714 102.159 300.737 102.159C298.305 102.159 296.703 104.011 296.703 106.693C296.703 109.403 298.283 111.176 300.857 111.176ZM298.402 105.875C298.47 104.636 299.368 103.562 300.749 103.562C302.067 103.562 302.93 104.54 302.936 105.875H298.402ZM313.026 104.403C312.673 103.045 311.611 102.159 309.724 102.159C307.753 102.159 306.355 103.199 306.355 104.744C306.355 105.983 307.105 106.807 308.741 107.17L310.219 107.494C311.06 107.682 311.452 108.057 311.452 108.602C311.452 109.278 310.73 109.807 309.616 109.807C308.599 109.807 307.946 109.369 307.741 108.511L306.099 108.761C306.384 110.307 307.668 111.176 309.628 111.176C311.736 111.176 313.196 110.057 313.196 108.477C313.196 107.244 312.412 106.483 310.81 106.114L309.423 105.795C308.463 105.568 308.048 105.244 308.054 104.653C308.048 103.983 308.776 103.506 309.741 103.506C310.798 103.506 311.287 104.091 311.486 104.676L313.026 104.403Z"
        fill="white"
      />
    </a>
  </g>
)

export default Aggregates
