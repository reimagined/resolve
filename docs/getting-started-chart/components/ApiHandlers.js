import React, { useState } from 'react'

const ApiHandlers = ({ selected, onClick }) => {
  return (
    <g
      id="interactscm-apihandlers"
      className="box-solid box-interactive"
      data-selected={selected}
    >
      <a onClick={onClick}>
        <path
          d="M46.718 161.99H61.723000000000006V184.172H46.718z"
          paintOrder="fill markers stroke"
        ></path>
        <text
          x="54.158"
          y="172.397"
          strokeWidth="0.265"
          fontFamily="sans-serif"
          fontSize="2.559"
          letterSpacing="0"
          textAnchor="middle"
          wordSpacing="0"
          style={{ lineHeight: '1.25' }}
        >
          <tspan
            x="54.158"
            y="172.397"
            style={{
              fontFeatureSettings: 'normal',
              fontVariantCaps: 'normal',
              fontVariantLigatures: 'normal',
              fontVariantNumeric: 'normal',
            }}
          >
            API
          </tspan>
          <tspan
            x="54.158"
            y="175.596"
            style={{
              fontFeatureSettings: 'normal',
              fontVariantCaps: 'normal',
              fontVariantLigatures: 'normal',
              fontVariantNumeric: 'normal',
            }}
          >
            Handlers
          </tspan>
        </text>
      </a>
    </g>
  )
}

export default ApiHandlers
