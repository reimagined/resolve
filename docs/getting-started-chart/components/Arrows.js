import React, { useState } from 'react'

const Arrows = () => (
  <g
    id="interactscm-arrows"
    className="arrows"
    paintOrder="fill markers stroke"
  >
    <path d="M201.939 164.061C202.525 164.646 203.475 164.646 204.061 164.061L213.607 154.515C214.192 153.929 214.192 152.979 213.607 152.393C213.021 151.808 212.071 151.808 211.485 152.393L203 160.879L194.515 152.393C193.929 151.808 192.979 151.808 192.393 152.393C191.808 152.979 191.808 153.929 192.393 154.515L201.939 164.061ZM204.5 136C204.5 135.172 203.828 134.5 203 134.5C202.172 134.5 201.5 135.172 201.5 136H204.5ZM204.5 163V136H201.5V163H204.5Z" />
    <path d="M201.939 308.061C202.525 308.646 203.475 308.646 204.061 308.061L213.607 298.515C214.192 297.929 214.192 296.979 213.607 296.393C213.021 295.808 212.071 295.808 211.485 296.393L203 304.879L194.515 296.393C193.929 295.808 192.979 295.808 192.393 296.393C191.808 296.979 191.808 297.929 192.393 298.515L201.939 308.061ZM204.5 280C204.5 279.172 203.828 278.5 203 278.5C202.172 278.5 201.5 279.172 201.5 280H204.5ZM204.5 307V280H201.5V307H204.5Z" />
    <path d="M412.939 86.9393C412.354 87.5251 412.354 88.4749 412.939 89.0607L422.485 98.6066C423.071 99.1924 424.021 99.1924 424.607 98.6066C425.192 98.0208 425.192 97.0711 424.607 96.4853L416.121 88L424.607 79.5147C425.192 78.9289 425.192 77.9792 424.607 77.3934C424.021 76.8076 423.071 76.8076 422.485 77.3934L412.939 86.9393ZM463 89.5C463.828 89.5 464.5 88.8284 464.5 88C464.5 87.1716 463.828 86.5 463 86.5V89.5ZM414 89.5H463V86.5H414V89.5Z" />
    <path d="M414 362.5C413.172 362.5 412.5 363.172 412.5 364C412.5 364.828 413.172 365.5 414 365.5V362.5ZM464.061 365.061C464.646 364.475 464.646 363.525 464.061 362.939L454.515 353.393C453.929 352.808 452.979 352.808 452.393 353.393C451.808 353.979 451.808 354.929 452.393 355.515L460.879 364L452.393 372.485C451.808 373.071 451.808 374.021 452.393 374.607C452.979 375.192 453.929 375.192 454.515 374.607L464.061 365.061ZM414 365.5H463V362.5H414V365.5Z" />
    <path d="M414 209.5C413.172 209.5 412.5 210.172 412.5 211C412.5 211.828 413.172 212.5 414 212.5V209.5ZM464.061 212.061C464.646 211.475 464.646 210.525 464.061 209.939L454.515 200.393C453.929 199.808 452.979 199.808 452.393 200.393C451.808 200.979 451.808 201.929 452.393 202.515L460.879 211L452.393 219.485C451.808 220.071 451.808 221.021 452.393 221.607C452.979 222.192 453.929 222.192 454.515 221.607L464.061 212.061ZM414 212.5H463V209.5H414V212.5Z" />
    <path d="M412.939 231.939C412.354 232.525 412.354 233.475 412.939 234.061L422.485 243.607C423.071 244.192 424.021 244.192 424.607 243.607C425.192 243.021 425.192 242.071 424.607 241.485L416.121 233L424.607 224.515C425.192 223.929 425.192 222.979 424.607 222.393C424.021 221.808 423.071 221.808 422.485 222.393L412.939 231.939ZM463 234.5C463.828 234.5 464.5 233.828 464.5 233C464.5 232.172 463.828 231.5 463 231.5V234.5ZM414 234.5H463V231.5H414V234.5Z" />
    <path d="M79 209.5C78.1716 209.5 77.5 210.172 77.5 211C77.5 211.828 78.1716 212.5 79 212.5V209.5ZM129.061 212.061C129.646 211.475 129.646 210.525 129.061 209.939L119.515 200.393C118.929 199.808 117.979 199.808 117.393 200.393C116.808 200.979 116.808 201.929 117.393 202.515L125.879 211L117.393 219.485C116.808 220.071 116.808 221.021 117.393 221.607C117.979 222.192 118.929 222.192 119.515 221.607L129.061 212.061ZM79 212.5H128V209.5H79V212.5Z" />
    <path d="M77.9393 231.939C77.3536 232.525 77.3536 233.475 77.9393 234.061L87.4853 243.607C88.0711 244.192 89.0208 244.192 89.6066 243.607C90.1924 243.021 90.1924 242.071 89.6066 241.485L81.1213 233L89.6066 224.515C90.1924 223.929 90.1924 222.979 89.6066 222.393C89.0208 221.808 88.0711 221.808 87.4853 222.393L77.9393 231.939ZM128 234.5C128.828 234.5 129.5 233.828 129.5 233C129.5 232.172 128.828 231.5 128 231.5V234.5ZM79 234.5H128V231.5H79V234.5Z" />
  </g>
)

export default Arrows
