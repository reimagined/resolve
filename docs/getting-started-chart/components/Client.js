import React, { useState } from 'react'

const Client = ({ selected, onClick }) => (
  <g
    id="interactscm-client"
    className="box-framed box-interactive"
    data-selected={selected}
  >
    <a onClick={onClick}>
      <path
        strokeLinejoin="round"
        strokeWidth="0.5"
        d="M70.903 152.15H82.887V194.938H70.903z"
        paintOrder="fill markers stroke"
      ></path>
      <text
        x="71.818"
        y="173.186"
        fill="#3949ab"
        strokeWidth="0.087"
        fontFamily="sans-serif"
        fontSize="3.468"
        letterSpacing="0"
        wordSpacing="0"
        style={{ lineHeight: '1.25' }}
      >
        <tspan x="72.2" y="173.186">
          Client
        </tspan>
      </text>
    </a>
  </g>
)

export default Client
