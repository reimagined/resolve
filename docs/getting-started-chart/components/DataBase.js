import React from 'react'

const DataBase = () => (
  <g className="db-icon" paintOrder="fill markers stroke">
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M64 230V220.065C58.2743 224.787 46.9875 228 34 228C21.0125 228 9.72571 224.787 4 220.065V230C4 230.759 4.33205 231.79 5.55928 233.07C6.80569 234.371 8.80022 235.724 11.5729 236.947C17.1074 239.389 25.0334 241 34 241C42.9666 241 50.8926 239.389 56.4271 236.947C59.1998 235.724 61.1943 234.371 62.4407 233.07C63.6679 231.79 64 230.759 64 230ZM68 230V213V196C68 187.716 52.7777 181 34 181C15.2223 181 0 187.716 0 196V213V230V247C0 255.284 15.2223 262 34 262C52.7777 262 68 255.284 68 247V230ZM64 237.065C58.2743 241.787 46.9875 245 34 245C21.0125 245 9.72571 241.787 4 237.065V247C4 247.759 4.33205 248.79 5.55928 250.07C6.80569 251.371 8.80022 252.724 11.5729 253.947C17.1074 256.389 25.0334 258 34 258C42.9666 258 50.8926 256.389 56.4271 253.947C59.1998 252.724 61.1943 251.371 62.4407 250.07C63.6679 248.79 64 247.759 64 247V237.065ZM4 203.065V213C4 213.759 4.33205 214.79 5.55928 216.07C6.80569 217.371 8.80022 218.724 11.5729 219.947C17.1074 222.389 25.0334 224 34 224C42.9666 224 50.8926 222.389 56.4271 219.947C59.1998 218.724 61.1943 217.371 62.4407 216.07C63.6679 214.79 64 213.759 64 213V203.065C58.2743 207.787 46.9875 211 34 211C21.0125 211 9.72571 207.787 4 203.065ZM4 196C4 196.759 4.33205 197.79 5.55928 199.07C6.80569 200.371 8.80022 201.724 11.5729 202.947C17.1074 205.389 25.0334 207 34 207C42.9666 207 50.8926 205.389 56.4271 202.947C59.1998 201.724 61.1943 200.371 62.4407 199.07C63.6679 197.79 64 196.759 64 196C64 195.241 63.6679 194.21 62.4407 192.93C61.1943 191.629 59.1998 190.276 56.4271 189.053C50.8926 186.611 42.9666 185 34 185C25.0334 185 17.1074 186.611 11.5729 189.053C8.80022 190.276 6.80569 191.629 5.55928 192.93C4.33205 194.21 4 195.241 4 196Z"
    />
  </g>
)

export default DataBase
