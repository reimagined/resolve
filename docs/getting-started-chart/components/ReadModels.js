import React, { useState } from 'react'

const ReadModels = () => (
  <g className="box-child-framed" data-selected="false">
    <path
      fill="#dae1f6"
      d="M27.095 191.13H37.736V196.995H27.095z"
      paintOrder="fill markers stroke"
    ></path>
    <text
      x="32.366"
      y="193.545"
      fill="#3949ab"
      strokeWidth="0.265"
      fontFamily="sans-serif"
      fontSize="2.056"
      letterSpacing="0"
      textAnchor="middle"
      wordSpacing="0"
      style={{ lineHeight: '1.25' }}
    >
      <tspan
        x="32.366"
        y="193.545"
        style={{
          fontFeatureSettings: 'normal',
          fontVariantCaps: 'normal',
          fontVariantLigatures: 'normal',
          fontVariantNumeric: 'normal',
        }}
      >
        Read
      </tspan>
      <tspan
        x="32.366"
        y="196.114"
        style={{
          fontFeatureSettings: 'normal',
          fontVariantCaps: 'normal',
          fontVariantLigatures: 'normal',
          fontVariantNumeric: 'normal',
        }}
      >
        Models
      </tspan>
    </text>
  </g>
)

export default ReadModels
