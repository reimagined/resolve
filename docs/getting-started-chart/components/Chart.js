import React, { useState } from 'react'

import Adapter from './Adapter'
import Aggregates from './Aggregates'
import ApiHandlers from './ApiHandlers'
import Arrows from './Arrows'
import Client from './Client'
import DataBase from './DataBase'
import ReadModels from './ReadModels'
import ReadSide from './ReadSide'
import Sagas from './Sagas'
import Server from './Server'
import ViewModels from './ViewModels'
import WriteSide from './WriteSide'

import AggregatesPage from '../pages/_aggregates.mdx'
import ApiHandlersPage from '../pages/_api-handlers.mdx'
import ClientPage from '../pages/_client.mdx'
import EsAdapterPage from '../pages/_es-adapter.mdx'
import ReadModelsPage from '../pages/_read-models.mdx'
import ReadSidePage from '../pages/_read-side.mdx'
import SagasPage from '../pages/_sagas.mdx'
import ServerPage from '../pages/_server.mdx'
import ViewModelsPage from '../pages/_view-models.mdx'
import WriteSidePage from '../pages/_write-side.mdx'

import '../static/styles.css'

const getPartial = (tab) => {
  switch (tab) {
    case 'aggregates':
      return <AggregatesPage></AggregatesPage>
    case 'apiHandlers':
      return <ApiHandlersPage></ApiHandlersPage>
    case 'client':
      return <ClientPage></ClientPage>
    case 'esAdapter':
      return <EsAdapterPage></EsAdapterPage>
    case 'readModels':
      return <ReadModelsPage></ReadModelsPage>
    case 'readSide':
      return <ReadSidePage></ReadSidePage>
    case 'sagas':
      return <SagasPage></SagasPage>
    case 'server':
      return <ServerPage></ServerPage>
    case 'viewModels':
      return <ViewModelsPage></ViewModelsPage>
    case 'writeSide':
      return <WriteSidePage></WriteSidePage>
  }
}

const renderDetail = (tab) => (
  <div className="alert alert--info">{getPartial(tab)}</div>
)

const Chart = ({ selected, onClick }) => {
  const [selectedTab, setSelectedTab] = useState(null)

  const getSelected = (tabName) => tabName == selectedTab

  return (
    <div>
      <svg
        width="550"
        height="445"
        viewBox="0 0 550 445"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <Server
            selected={getSelected('server')}
            onClick={() => setSelectedTab('server')}
          />
          <WriteSide
            selected={getSelected('writeSide')}
            onClick={() => setSelectedTab('writeSide')}
          />
          <Aggregates
            selected={getSelected('aggregates')}
            onClick={() => setSelectedTab('aggregates')}
          />
          <DataBase />
          <ApiHandlers
            selected={getSelected('apiHandlers')}
            onClick={() => setSelectedTab('apiHandlers')}
          />
          <Adapter
            selected={getSelected('esAdapter')}
            onClick={() => setSelectedTab('esAdapter')}
          />
          <ReadSide
            selected={getSelected('readSide')}
            onClick={() => setSelectedTab('readSide')}
          />
          <ReadModels
            selected={getSelected('readModels')}
            onClick={() => setSelectedTab('readModels')}
          />
          <Sagas
            selected={getSelected('sagas')}
            onClick={() => setSelectedTab('sagas')}
          />
          <ViewModels
            selected={getSelected('viewModels')}
            onClick={() => setSelectedTab('viewModels')}
          />

          <Client
            selected={getSelected('client')}
            onClick={() => setSelectedTab('client')}
          />
          <Arrows />
        </g>
      </svg>
      <div>{renderDetail(selectedTab)}</div>
    </div>
  )
}

export default Chart
