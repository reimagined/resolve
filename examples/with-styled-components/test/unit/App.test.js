import React from 'react'
import renderer from 'react-test-renderer'

import { StyledComponents } from '../../client/components/StyledComponents'

test('renders correctly', () => {
  const tree = renderer.create(<StyledComponents />).toJSON()
  expect(tree).toMatchSnapshot()
})
