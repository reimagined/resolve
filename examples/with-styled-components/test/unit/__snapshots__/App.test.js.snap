// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-bwzfXH dabeYK"
>
  <h1
    className="sc-bdVaJa bzmvhR"
  >
    Hello World, this is my first styled component!
  </h1>
  <h1
    className="sc-bdVaJa bzmvhR"
  >
    SSR works too!
  </h1>
</section>
`;
